src/components/ShoppingList/ShoppingList.js

```diff
@@ -62,6 +62,90 @@ function ShoppingList({ listId }) {
     setFilter(e.target.value);
   };
 
+  const getDaysToPurchase = (item) => {
+    let nextPurchaseDate;
+    if (item.lastPurchaseDate?.seconds) {
+      // if the item has been purchased before, next purchase date is [purchaseInterval] days from the lastPurchaseDate
+      nextPurchaseDate = DateTime.fromSeconds(
+        item.lastPurchaseDate.seconds,
+      ).plus({ days: item.purchaseInterval });
+    } else if (item.createdAt?.seconds) {
+      // if there's no purchase history, estimate it will be bought [purchaseInterval] days from when item was created
+      // (user provides this info at item creation)
+      nextPurchaseDate = DateTime.fromSeconds(item.createdAt.seconds).plus({
+        days: item.purchaseInterval,
+      });
+    } else return null;
+
+    const currentDate = DateTime.fromSeconds(Math.floor(Date.now() / 1000));
+    const daysRemaining = nextPurchaseDate.diff(currentDate, ['days']);
+    return Math.round(daysRemaining.as('days'));
+  };
+
+  const isItemInactive = (item) => {
+    const currentDate = DateTime.fromSeconds(Math.floor(Date.now() / 1000));
+
+    if (item.lastPurchaseDate?.seconds) {
+      // if an item has been purchased before, compare the time elapsed since lastPurchaseDate to the purchaseInterval
+      return (
+        currentDate
+          .diff(DateTime.fromSeconds(item.lastPurchaseDate.seconds), ['days'])
+          .as('days') >=
+        2 * item.purchaseInterval
```

Comment:
Ah, I see it in your PR. Again, It would be beneficial to add what you have in the PR as a comment here, especially for the long-term so we remember why we multiplied by 2!

Reply:
Ah, yes it's in the AC that this is how inactive items are defined. Should definitely add a comment in here, thanks!